src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterTextViewModelProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation.IntellisenseControls;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Projection;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature
+{
+    [Export(typeof(ITextViewModelProvider))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(VisualStudioChangeSignatureOptionsService.AddParameterTextViewRole)]
+    internal class AddParameterTextViewModelProvider : ITextViewModelProvider
```

Comment:
This PR is great fun because there are classes with 'ViewModel' in the name that are ITextViewModels, and things which are WPF-style ViewModels. This has confused more than once. I think you're a victim of circumstance though!